### PR TITLE
fix: exhaustive deps lint errors

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

Fixed the remaining eslint warnings (the vast majority of which were `exhaustive-deps`) and made our rules more strict going forward. This involved a lot `useCallback` updates to prevent infinite re-renders and some minor code changes that don't change logic in a few components to use primitives rather than objects. In some cases, I think it is a code smell that we have so many dependencies in our `useEffect`s, but I didn't want to change much besides fixing the lint warnings in place. One PR at a time. The worst offender of this is `Splash.tsx` with 18(!) dependencies in one case.

There is one snapshot change, but just because I added more data to the test messages in `Chat.test.tsx`

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/components/misc/NewQRView.tsx [hunk 1]
```diff
@@ -130,53 +130,50 @@ const NewQRView: React.FC<Props> = ({ defaultToConnect, handleCodeScan, error, e
...
-      }}
-      icon="share-variant"
-    />
-  )
+  }, [navigation])
 
   useEffect(() => {
-    let headerRight = invitation ? scanShareUrl : undefined
+    let headerRight = invitation ? (
+      <HeaderButton
```
Comment (jleach) [inline packages/legacy/core/App/components/misc/NewQRView.tsx:141]: Curious why these needed to go inside the `useEffect`? Seems odd to have a component defined in a `useEffect` but I can't think of any good reason not to do it - just what I'm used to I suppose.

File: packages/legacy/core/App/components/misc/NewQRView.tsx [hunk 2]
```diff
@@ -130,53 +130,50 @@ const NewQRView: React.FC<Props> = ({ defaultToConnect, handleCodeScan, error, e
...
-        Share.share({ message: invitation ?? '' })
-      }}
-      icon="share-variant"
-    />
-  )
+  }, [navigation])
 
   useEffect(() => {
-    let headerRight = invitation ? scanShareUrl : undefined
+    let headerRight = invitation ? (
```
Comment (jleach) [inline packages/legacy/core/App/components/misc/NewQRView.tsx:140]: Wonder if we can throw a space in this fn to make it a little easier to read. // wall-of-text.

**Review by jleach — CHANGES_REQUESTED** [review]

File: packages/legacy/core/App/components/misc/NewQRView.tsx (hunk 1 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/misc/NewQRView.tsx:141]: If they are just in the component, they will get redefined every time it renders. We are only using them when setting the navigation options when the tab changes, so we watch for that and only change them then. Maybe not the best way to do it, but better than it was.

File: packages/legacy/core/App/components/misc/NewQRView.tsx (hunk 2 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/components/misc/NewQRView.tsx:140]: done

**Review by jleach — APPROVED** [review]

**Pull request merged**